src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -429,8 +429,8 @@ Copyright (c) .NET Foundation. All rights reserved.
     -->
   <Target Name="_CreateR2RSymbols"
           Inputs="@(_ReadyToRunSymbolsCompileList)"
-          Outputs="%(_ReadyToRunSymbolsCompileList.OutputPDBImage)">
-
+          Outputs="%(_ReadyToRunSymbolsCompileList.OutputPDBImage)"
+          Condition="'$(PublishReadyToRunUseCrossgen2)' != 'true'">
```

Comment:
This isn't correct for crossgen2 on .NET 5.0. We should continue to create pdbs with crossgen when compiling targetting .NET 5.0 or earlier.

Reply:
Fixed in 3rd commit, thanks for pointing that out!